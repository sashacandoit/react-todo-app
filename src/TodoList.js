import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo"
import { v4 as uuid } from "uuid";

const TodoList = () => {
    const INITIAL_STATE = [
        {id: uuid(), todo: 'Make first todo!', completed: false}
    ]
    const [todos, setTodos] = useState(INITIAL_STATE);
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, { ...newTodo, id: uuid() }]);
    }

    return (
        <div>
            <h1>React Todo List</h1>
            {/* <NewTodoForm addTodo={addTodo} /> */}
            <ol>
                {todos.map(({id, todo, completed}) => <Todo id={id} todo={todo} key={id} completed={completed} />)}
            </ol>
        </div>
    )
}

export default TodoList;