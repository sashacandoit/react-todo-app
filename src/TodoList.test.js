import React from "react";
import { render } from "@testing-library/react";
import TodoList from "./TodoList"

//smoke test
it("renders without crashing", function () {
    render(<TodoList />);
});

//snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});