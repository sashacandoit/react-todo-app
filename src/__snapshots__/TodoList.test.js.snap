// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1>
      React Todo List
    </h1>
    <form
      class="NewTodoForm"
    >
      <span
        class="NewTodoForm-inputGroup"
      >
        <label
          for="todo"
        >
          Todo: 
        </label>
        <input
          id="todo"
          name="todo"
          placeholder="Something that needs to get done"
          type="text"
          value=""
        />
      </span>
      <span
        class="NewTodoForm-button"
      >
        <button>
          Add
        </button>
      </span>
    </form>
    <ol>
      <li
        class="Todo"
      >
        Make first todo!
        <span
          class="Todo-delete"
        >
          X
        </span>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
